snapcraft/projects.py

```diff
@@ -47,23 +47,43 @@ class Config:  # pylint: disable=too-few-public-methods
 # see https://github.com/samuelcolvin/pydantic/issues/975#issuecomment-551147305
 # fmt: off
 if TYPE_CHECKING:
-    CommandChainStr = str
     UniqueStrList = List[str]
-    UniqueAliasList = List[str]
 else:
-    CommandChainStr = constr(regex=r"^[A-Za-z0-9/._#:$-]*$")
     UniqueStrList = conlist(str, unique_items=True)
-    UniqueAliasList = conlist(constr(regex=r"^[a-zA-Z0-9][-_.a-zA-Z0-9]*$"), unique_items=True)
```

Comment:
Catching errors with `@pydantic.validator` functions makes for more descriptive errors.